src/main/java/com/jfrog/tasks/GenerateDepTrees.java

```diff
@@ -169,8 +169,11 @@ private GradleDepTreeResults createProjectDependencyTree(Project project) {
         Map<String, GradleDependencyNode> nodes = new HashMap<>();
         nodes.put(rootId, root);
 
-        for (Configuration configuration : project.getConfigurations()) {
-            addConfiguration(root, configuration, nodes);
+        // getConfigurations() may throw java.util.ConcurrentModificationException in loop
```

Comment:
The implementation looks good. I just think this comment is not clear enough.
There's another suggestion:
```
// To prevent ConcurrentModificationException, we clone the configuration names before iterating over them.
// This avoids issues caused by dynamic modifications by other Gradle plugins.
```